### ISSUE TITLE
Config warning in simple reactor example

### ISSUE THREAD

**Issue by hsaunders1904** [Fusion-Power-Plant-Framework/bluemira#2280]
## Describe the bug

When you run the reactor example, you get the below warning:

```
+-------------------------------------------------------------------------+
| WARNING: Empty config for args: ('Plasma', 'designer')                  |
+-------------------------------------------------------------------------+
```

I'm of the opinion that, really, we shouldn't have any warnings in our examples (unless intentional). Especially for such an important introductory example, it's like we're indicating that there's something wrong.

Also, not sure that I'm a big fan of the warning in general. Some designers/builders may not have any config options, or will mostly be used with their defaults; in the former case, there's no way to stop the warning by editing the config. I see there's a `warn_on_empty_config` option in the `ReactorConfig` constructor. I wonder if this should be `False` by default?

## Steps to reproduce

Run the simple reactor example

```console
python bluemira/base/reactor_config.py
```

## Expected behaviour

No warnings in the example's output.

**Comment by je-cook** [comment]
Yeah, I agree although I think if the warning is default off we should consider making them debug messages instead of completely disabling.

**Comment by je-cook** [comment]
yeah exactly that

**Comment by oliverfunk** [comment]
I've gone in a few loops in my head over this and I think it's better to leave it as but set *all* the enable warnings to False by default, because:

* warn_on_empty_config: You'll quickly realise if you have a spelling mistake in the args because your config is not pulling through. Being warned over and over isn't really the intention with the warning, more to let you know that you could have made a mistake because it's empty. So it should be off by default and enabled by the user if they really want the warning. A lot of development is done with logging set to DEBUG so ya then forcing it to log to debug even when it's disabled seems counterintuitive.

* warn_on_duplicate_keys: As a user, you've probably done this (constructed your config like this) intentionally and being warned repeatedly is only annoying and isn't the point again. If you've made constructed a config that has duplicated keys, you've probably thought about how those interact/override each other and will read the docs.

* warn_on_empty_local_params: Basically the same reasoning as above, you've internally left the local params empty because you're using only globals. Case and point is EUDEMO.

A big rant for warnings #crysis #devlife haha.

**Comment by je-cook** [comment]
I definitely think the last two should have debug logging when not warning because as much as you may expect the user to read the docs that's not exactly likely in most cases just judging by myself 😅. 

If we put it logging to debug at least its more likely an error will be found when trying to find errors by searching the log (the first port of call for a user debugging).

I would suggest that even in the first case that is also true. Some small (typing) error could easily fall through the gaps that may not be noticed.

Personally I don't develop with debug on unless I've broken something.
In my head debug is meant to be noisy but still decipherable whereas if we were to say add a TRACE level it would be a firehose of data. I don't think this is at the level. 
I also dont think its counter intuitive to still log to debug if false we are just turning on/off warnings not on/off logging.

As a side point our NOT_SET level (essentially TRACE) should probably be looped into the logging printing stuff but maybe that should happen when we clean up the logging backend stuff. For instance all logs currently get dumped into `bluemira.log` irrelevant of your current log level and its not (easily) configurable.

 Maybe @hsaunders1904 can weigh in too.

**Comment by je-cook** [comment]
exactly yeah
